wasm-engines/Dockerfile

```diff
@@ -0,0 +1,221 @@
+RUN pip3 install click durationpy
+
+
+# install nodejs
+#RUN mkdir node
+RUN mkdir -p node
+RUN cd node && curl -fsSLO --compressed https://nodejs.org/dist/v11.10.0/node-v11.10.0-linux-x64.tar.gz
+RUN cd node && tar -xvf node-v11.10.0-linux-x64.tar.gz -C /usr/local/ --strip-components=1 --no-same-owner
+RUN cd node && ln -s /usr/local/bin/node ./node
+COPY node-timer.js ./node/node-timer.js
+
+# curl -fsSLO --compressed https://nodejs.org/download/nightly/v12.0.0-nightly20190325cdb87d9548/node-v12.0.0-nightly20190325cdb87d9548-linux-x64.tar.gz
+
+
+# install java
+# ENV JAVA_VER 8
+# ENV JAVA_HOME /usr/lib/jvm/java-8-oracle
+
+# RUN \
+#  echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | debconf-set-selections && \
+#  add-apt-repository -y ppa:webupd8team/java && \
+#  apt-get update && \
+#  apt-get install -y oracle-java8-installer && \
+#  rm -rf /var/lib/apt/lists/* && \
+#  rm -rf /var/cache/oracle-jdk8-installer
+
+# ENV JAVA_HOME /usr/lib/jvm/java-8-oracle
+
+
+#RUN echo 'deb http://ppa.launchpad.net/webupd8team/java/ubuntu trusty main' >> /etc/apt/sources.list && \
+#    echo 'deb-src http://ppa.launchpad.net/webupd8team/java/ubuntu trusty main' >> /etc/apt/sources.list 
+
+#RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys C2518248EEA14886
+#RUN apt-get update
+#RUN echo oracle-java${JAVA_VER}-installer shared/accepted-oracle-license-v1-1 select true | sudo /usr/bin/debconf-set-selections 
+
+#RUN apt-get install -y --force-yes --no-install-recommends oracle-java${JAVA_VER}-installer oracle-java${JAVA_VER}-set-default 
+#RUN apt-get clean 
+#RUN rm -rf /var/cache/oracle-jdk${JAVA_VER}-installer
+
+# install asmble
+#RUN wget https://github.com/cdetrio/asmble/releases/download/0.4.2-fl-bench-times/asmble-0.4.2-fl-bench-times.tar
+#RUN tar -xvf asmble-0.4.2-fl-bench-times.tar
+
+# install wag
+#RUN apt-get install -y libcapstone-dev
+#RUN git clone --single-branch --branch runtime-design https://github.com/gballet/wag.git
+#RUN cd wag && go build ./cmd/ethereum/...
+
+# install wamr (interpreter, aot, jit)
+RUN git clone https://github.com/hugo-dc/wasm-micro-runtime.git --branch benchmark
+RUN cd wasm-micro-runtime && git pull origin benchmark
+## Build LLVM
+RUN cd wasm-micro-runtime/product-mini/platforms/linux && ./build_llvm.sh 
+## Build JIT
+RUN cd wasm-micro-runtime/product-mini/platforms/linux && ./build_jit.sh 
+## Build Interpreter
+RUN cd wasm-micro-runtime/product-mini/platforms/linux && mkdir build_interp && cd build_interp && cmake -DWAMR_BUILD_INTERP=1 .. -DCMAKE_BUILD_TYPE=Release .. && make
+## Build Compiler
+RUN cd wasm-micro-runtime/wamr-compiler && mkdir build && cd build && cmake -DCMAKE_BUILD_TYPE=Release .. && make
+
+# install wasm3
+RUN git clone https://github.com/hugo-dc/wasm3.git --branch benchmark
+RUN cd wasm3 && mkdir build && cd build && cmake -DCMAKE_BUILD_TYPE=Release .. && make
+
+# Update gcc/g++
+RUN apt install -y gcc-8
```

Comment:
You should be using GCC 9 (the latest).

Reply:
Upgrading docker base image to Ubuntu 19.10 is probably the easiest way to get it.